roachtest: sqlsmith: tracking issue for inbox communication errors

roachtest.sqlsmith/setup=seed-vec/setting=vec [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3058191&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3058191&tab=artifacts#/sqlsmith/setup=seed-vec/setting=vec) on master @ [cef0bd947590218ea6ac94f4f85dffc25e16fcd0](https://github.com/cockroachdb/cockroach/commits/cef0bd947590218ea6ac94f4f85dffc25e16fcd0):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/sqlsmith/setup=seed-vec/setting=vec/run_1
	sqlsmith.go:221,sqlsmith.go:251,test_runner.go:801: error: pq: inbox communication error: rpc error: code = Canceled desc = context canceled
		stmt:
		SELECT
			tab_7694._bool AS col_28205,
			tab_7694._timestamptz AS col_28206,
			tab_7695._bool AS col_28207,
			tab_7696._int4 AS col_28208,
			tab_7694._float8 AS col_28209,
			tab_7694._interval AS col_28210,
			tab_7693._timestamp AS col_28211,
			tab_7696._bool AS col_28212,
			tab_7694._string AS col_28213
		FROM
			defaultdb.public.seed_vec@seed_vec__int8__float8__date_idx AS tab_7693,
			defaultdb.public.seed_vec@[0] AS tab_7694,
			defaultdb.public.seed_vec@seed_vec__int8__float8__date_idx AS tab_7695
			JOIN defaultdb.public.seed_vec@seed_vec__int8__float8__date_idx AS tab_7696 ON
					(tab_7695._int2) = (tab_7696._int4)
					AND (tab_7695._uuid) = (tab_7696._uuid)
					AND (tab_7695._timestamp) = (tab_7696._timestamp)
					AND (tab_7695._int8) = (tab_7696._int8)
		WHERE
			tab_7696._bool
		ORDER BY
			tab_7693._int2
		LIMIT
			8:::INT8;
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
# From https://go.crdb.dev/p/roachstress, perhaps edited lightly.
caffeinate ./roachstress.sh sqlsmith/setup=seed-vec/setting=vec
```
</p>
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*sqlsmith/setup=seed-vec/setting=vec.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.sqlsmith/setup=seed-vec/setting=vec [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3123385&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3123385&tab=artifacts#/sqlsmith/setup=seed-vec/setting=vec) on master @ [a142e392ddebb2d8a7c575170ac168cdefa90088](https://github.com/cockroachdb/cockroach/commits/a142e392ddebb2d8a7c575170ac168cdefa90088):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/sqlsmith/setup=seed-vec/setting=vec/run_1
	sqlsmith.go:222,sqlsmith.go:252,test_runner.go:758: error: pq: inbox communication error: rpc error: code = Canceled desc = context canceled
		stmt:
		SELECT
			tab_150778._int4 AS col_550808,
			tab_150779._string AS col_550809,
			tab_150778._timestamp AS col_550810,
			tab_150778._date AS col_550811,
			tab_150779._bytes AS col_550812
		FROM
			defaultdb.public.seed_vec@seed_vec__int8__float8__date_idx AS tab_150778,
			defaultdb.public.seed_vec@[0] AS tab_150779,
			defaultdb.public.seed_vec AS tab_150780
			JOIN defaultdb.public.seed_vec AS tab_150781 ON
					(tab_150780._int8) = (tab_150781._int8)
					AND (tab_150780._date) = (tab_150781._date)
					AND (tab_150780._float8) = (tab_150781._float8)
		WHERE
			tab_150779._bool
		ORDER BY
			tab_150778._timestamptz,
			tab_150778._bool,
			tab_150778._float8 ASC,
			tab_150779._decimal ASC,
			tab_150779._int8 ASC,
			tab_150780._float8
		LIMIT
			31:::INT8;
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
# From https://go.crdb.dev/p/roachstress, perhaps edited lightly.
caffeinate ./roachstress.sh sqlsmith/setup=seed-vec/setting=vec
```
</p>
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*sqlsmith/setup=seed-vec/setting=vec.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.sqlsmith/setup=seed-vec/setting=vec [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3135493&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3135493&tab=artifacts#/sqlsmith/setup=seed-vec/setting=vec) on master @ [ab6096559762862eb632877e8d94a4e2b5b10c50](https://github.com/cockroachdb/cockroach/commits/ab6096559762862eb632877e8d94a4e2b5b10c50):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/sqlsmith/setup=seed-vec/setting=vec/run_1
	sqlsmith.go:222,sqlsmith.go:252,test_runner.go:758: error: pq: inbox communication error: rpc error: code = Canceled desc = context canceled
		stmt:
		SELECT
			tab_166734._int8 AS col_615011,
			tab_166736._float8 AS col_615012,
			tab_166734._bool AS col_615013,
			tab_166734._string AS col_615014,
			tab_166733._float8 AS col_615015,
			tab_166733._float8 AS col_615016,
			tab_166735._float8 AS col_615017,
			tab_166735._float8 AS col_615018,
			tab_166734._date AS col_615019,
			tab_166734._interval AS col_615020,
			tab_166734._timestamptz AS col_615021,
			tab_166734.crdb_internal_mvcc_timestamp AS col_615022,
			tab_166735._float8 AS col_615023,
			tab_166734._float8 AS col_615024,
			tab_166734._int2 AS col_615025,
			tab_166734._int8 AS col_615026
		FROM
			defaultdb.public.seed_vec AS tab_166732
			JOIN defaultdb.public.seed_vec AS tab_166733 ON
					(tab_166732._int8) = (tab_166733._int8)
					AND (tab_166732._date) = (tab_166733._date)
					AND (tab_166732._float8) = (tab_166733._float8)
			JOIN defaultdb.public.seed_vec@[0] AS tab_166734 ON
					(tab_166733._float8) = (tab_166734._float8) AND (tab_166733._float8) = (tab_166734._float8),
			defaultdb.public.seed_vec AS tab_166735
			JOIN defaultdb.public.seed_vec AS tab_166736 ON
					(tab_166735._int8) = (tab_166736._int8) AND (tab_166735._date) = (tab_166736._date)
		WHERE
			tab_166734._bool
		ORDER BY
			tab_166734.tableoid, tab_166734._uuid, tab_166736._float8 DESC;
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
# From https://go.crdb.dev/p/roachstress, perhaps edited lightly.
caffeinate ./roachstress.sh sqlsmith/setup=seed-vec/setting=vec
```
</p>
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*sqlsmith/setup=seed-vec/setting=vec.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.sqlsmith/setup=seed-vec/setting=vec [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3146390&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3146390&tab=artifacts#/sqlsmith/setup=seed-vec/setting=vec) on master @ [eb7f3fe2991992472a68d701d65543f7a9c7fb56](https://github.com/cockroachdb/cockroach/commits/eb7f3fe2991992472a68d701d65543f7a9c7fb56):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/sqlsmith/setup=seed-vec/setting=vec/run_1
	sqlsmith.go:222,sqlsmith.go:252,test_runner.go:761: error: pq: inbox communication error: rpc error: code = Canceled desc = context canceled
		stmt:
		SELECT
			tab_86699.crdb_internal_mvcc_timestamp AS col_319590, tab_86701._float8 AS col_319591
		FROM
			defaultdb.public.seed_vec@seed_vec__int8__float8__date_idx AS tab_86699,
			defaultdb.public.seed_vec@seed_vec__int8__float8__date_idx AS tab_86700
			JOIN defaultdb.public.seed_vec@[0] AS tab_86701 ON (tab_86700._uuid) = (tab_86701._uuid)
		WHERE
			tab_86701._bool
		ORDER BY
			tab_86700._string ASC;
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
# From https://go.crdb.dev/p/roachstress, perhaps edited lightly.
caffeinate ./roachstress.sh sqlsmith/setup=seed-vec/setting=vec
```
</p>
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*sqlsmith/setup=seed-vec/setting=vec.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

Hm, the last failure here is concerning, node 1 crashed with
```
panic: runtime error: slice bounds out of range [:64] with capacity 57 [recovered]
	panic: runtime error: slice bounds out of range [:64] with capacity 57 [recovered]
	panic: runtime error: slice bounds out of range [:64] with capacity 57

goroutine 2672 [running]:
panic(0x4a18600, 0xc004818120)
	/usr/local/go/src/runtime/panic.go:1065 +0x565 fp=0xc001e3d498 sp=0xc001e3d3d0 pc=0x48b265
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).closeWrapper(0xc00138b000, 0x647fd48, 0xc001483200, 0x4bf6e00, 0xc001f4b6e0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:902 +0x3db fp=0xc001e3d528 sp=0xc001e3d498 pc=0x3305cdb
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1(0xc00138b000, 0x647fd48, 0xc001483200)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:624 +0x65 fp=0xc001e3d560 sp=0xc001e3d528 pc=0x34e8ee5
runtime.call32(0x0, 0x556f4d8, 0xc00237d188, 0x1800000018)
	/usr/local/go/src/runtime/asm_amd64.s:551 +0x3e fp=0xc001e3d590 sp=0xc001e3d560 pc=0x4c42fe
runtime.reflectcallSave(0xc001e3d6c8, 0x556f4d8, 0xc00237d188, 0xc000000018)
	/usr/local/go/src/runtime/panic.go:877 +0x58 fp=0xc001e3d5c0 sp=0xc001e3d590 pc=0x48ac38
runtime.runOpenDeferFrame(0xc0022e1500, 0xc00237d140, 0x0)
	/usr/local/go/src/runtime/panic.go:851 +0x62d fp=0xc001e3d648 sp=0xc001e3d5c0 pc=0x48a90d
panic(0x4bf6e00, 0xc001f4b6e0)
	/usr/local/go/src/runtime/panic.go:965 +0x1b9 fp=0xc001e3d710 sp=0xc001e3d648 pc=0x48aeb9
github.com/cockroachdb/cockroach/pkg/sql/flowinfra.(*FlowBase).Wait(0xc000edde40)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/flowinfra/flow.go:449 +0x1c7 fp=0xc001e3d7a8 sp=0xc001e3d710 pc=0x21c39e7
runtime.call16(0x0, 0x55708c8, 0xc0013ad268, 0x800000008)
	/usr/local/go/src/runtime/asm_amd64.s:550 +0x3e fp=0xc001e3d7c8 sp=0xc001e3d7a8 pc=0x4c425e
runtime.reflectcallSave(0xc001e3d900, 0x55708c8, 0xc0013ad268, 0x8)
	/usr/local/go/src/runtime/panic.go:877 +0x58 fp=0xc001e3d7f8 sp=0xc001e3d7c8 pc=0x48ac38
runtime.runOpenDeferFrame(0xc0022e1500, 0xc0013ad220, 0x0)
	/usr/local/go/src/runtime/panic.go:851 +0x62d fp=0xc001e3d880 sp=0xc001e3d7f8 pc=0x48a90d
panic(0x4bf6e00, 0xc001f4b6e0)
	/usr/local/go/src/runtime/panic.go:965 +0x1b9 fp=0xc001e3d948 sp=0xc001e3d880 pc=0x48aeb9
runtime.goPanicSliceAcap(0x40, 0x39)
	/usr/local/go/src/runtime/panic.go:106 +0xa5 fp=0xc001e3d990 sp=0xc001e3d948 pc=0x488985
github.com/cockroachdb/cockroach/pkg/col/coldata.(*Bytes).Get(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/col/coldata/bytes.go:122
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*writeBuffer).writeTextColumnarElement.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/types.go:319 +0xc15 fp=0xc001e3db38 sp=0xc001e3d990 pc=0x3a259b5
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*FmtCtx).WithDataConversionConfig(0xc002dc32d0, 0x200000000, 0xc000000000, 0xc001e3dbb0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/format.go:320 +0xb5 fp=0xc001e3db90 sp=0xc001e3db38 pc=0x17b5875
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*writeBuffer).writeTextColumnarElement(0xc001313a50, 0x649a8b8, 0xc0054030b0, 0x65c4ce8, 0xc004407f80, 0x3, 0x200000000, 0x0, 0x88dde00)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/types.go:288 +0xfe fp=0xc001e3dc10 sp=0xc001e3db90 pc=0x3a1c4be
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).bufferBatch(0xc001313200, 0x649a8b8, 0xc0054030b0, 0x65ae960, 0xc00317a420, 0x0, 0x0, 0x0, 0x200000000, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:1306 +0x412 fp=0xc001e3dce0 sp=0xc001e3dc10 pc=0x3a111d2
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*commandResult).AddBatch.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/command_result.go:231 +0xf8 fp=0xc001e3dd70 sp=0xc001e3dce0 pc=0x3a211b8
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*commandResult).addInternal(0xc001313898, 0xc001e3de00, 0xc0031d5b00, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/command_result.go:208 +0xc8 fp=0xc001e3dde0 sp=0xc001e3dd70 pc=0x3a08a68
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*commandResult).AddBatch(0xc001313898, 0x649a8b8, 0xc0054030b0, 0x65ae960, 0xc00317a420, 0xc001e3de90, 0x5570788)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/command_result.go:229 +0x87 fp=0xc001e3de40 sp=0xc001e3dde0 pc=0x3a08d67
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLReceiver).PushBatch(0xc004b13880, 0x65ae960, 0xc00317a420, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:1018 +0x122 fp=0xc001e3de88 sp=0xc001e3de40 pc=0x338f102
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*BatchFlowCoordinator).Run(0xc0008a4580, 0x647fd48, 0xc00489c200)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colflow/flow_coordinator.go:294 +0x21a fp=0xc001e3df98 sp=0xc001e3de88 pc=0x315fdda
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlow).Run(0xc0015d8190, 0x647fd48, 0xc00489c200, 0x556f468)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colflow/vectorized_flow.go:257 +0x234 fp=0xc001e3e030 sp=0xc001e3df98 pc=0x3164b54
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run(0xc001698780, 0xc0042d8700, 0xc0053bd220, 0xc00354f100, 0xc004b13880, 0xc00138b7a8, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:525 +0x4d7 fp=0xc001e3e668 sp=0xc001e3e030 pc=0x338cb57
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRun(0xc001698780, 0x649a8b8, 0xc0054030b0, 0xc00138b7a8, 0xc0042d8700, 0xc0053bd220, 0x6481ad0, 0xc0053bd400, 0x0, 0xc004b13880, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:1262 +0x1cb fp=0xc001e3e6e0 sp=0xc001e3e668 pc=0x3390a2b
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithDistSQLEngine(0xc00138b000, 0x649a8b8, 0xc0054030b0, 0xc00138b578, 0x3, 0x7fb902040df8, 0xc001313898, 0xc001e3e901, 0xc002747798, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1102 +0x468 fp=0xc001e3e808 sp=0xc001e3e6e0 pc=0x331aec8
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine(0xc00138b000, 0x649a8b8, 0xc0054030b0, 0xc00138b578, 0x7fb902040df8, 0xc001313898, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:964 +0x7bd fp=0xc001e3eaf8 sp=0xc001e3e808 pc=0x3319e3d
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState(0xc00138b000, 0x649a8b8, 0xc0054030b0, 0x64a7178, 0xc003f943c0, 0xc004c7d34e, 0xec, 0x0, 0x1, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:682 +0xfc6 fp=0xc001e3f4e0 sp=0xc001e3eaf8 pc=0x33167e6
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt(0xc00138b000, 0x647fd48, 0xc0031d5b00, 0x64a7178, 0xc003f943c0, 0xc004c7d34e, 0xec, 0x0, 0x1, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:126 +0xb8f fp=0xc001e3f748 sp=0xc001e3f4e0 pc=0x3314faf
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd.func1(0x64a7178, 0xc003f943c0, 0xc004c7d34e, 0xec, 0x0, 0x1, 0x350c93dd, 0xed88786b2, 0x0, 0x350ca493, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1618 +0x233 fp=0xc001e3f808 sp=0xc001e3f748 pc=0x34e9253
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd(0xc00138b000, 0x647fd48, 0xc0031d5b00, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1620 +0x1691 fp=0xc001e3fcc8 sp=0xc001e3f808 pc=0x3309771
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc00138b000, 0x647fd48, 0xc001483200, 0xc00169c320, 0x5400, 0x15000, 0xc00169c500, 0xc0006374d0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1542 +0x225 fp=0xc001e3fd68 sp=0xc001e3fcc8 pc=0x3307e65
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn(0xc0016a1700, 0x647fd48, 0xc001483200, 0xc00138b000, 0x5400, 0x15000, 0xc00169c500, 0xc0006374d0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:626 +0xce fp=0xc001e3fdf0 sp=0xc001e3fd68 pc=0x330398e
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync.func1(0xc0026d3f46, 0xc001254f48, 0x647fd48, 0xc001483200, 0xc0006374d0, 0xc0016a1700, 0xc001313200, 0x64e0ae8, 0xc001380280, 0xc002bf96e0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:645 +0x3f5 fp=0xc001e3ff38 sp=0xc001e3fdf0 pc=0x3a22f75
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1371 +0x1 fp=0xc001e3ff40 sp=0xc001e3ff38 pc=0x4c5cc1
created by github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:573 +0x185
```
However, the query contains a CASE operator which was recently refactored, and the SHA doesn't include the fix in #67757. I couldn't confirm it (nor repro the issue), but I'm hopeful that the issue is now fixed.

roachtest.sqlsmith/setup=seed-vec/setting=vec [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3259029&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3259029&tab=artifacts#/sqlsmith/setup=seed-vec/setting=vec) on master @ [eef03a46f2e43ff70485dadf7d9ad445db05cab4](https://github.com/cockroachdb/cockroach/commits/eef03a46f2e43ff70485dadf7d9ad445db05cab4):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/sqlsmith/setup=seed-vec/setting=vec/run_1
	sqlsmith.go:223,sqlsmith.go:252,test_runner.go:777: error: pq: inbox communication error: rpc error: code = Canceled desc = context canceled
		stmt:
		SELECT
			tab_142430._interval AS col_523726,
			tab_142429._int8 AS col_523727,
			tab_142428._timestamptz AS col_523728,
			tab_142429._timestamp AS col_523729,
			tab_142430._bool AS col_523730,
			tab_142431._date AS col_523731,
			tab_142430._string AS col_523732,
			tab_142428._string AS col_523733
		FROM
			defaultdb.public.seed_vec@seed_vec__int8__float8__date_idx AS tab_142428,
			defaultdb.public.seed_vec@seed_vec__int8__float8__date_idx AS tab_142429
			JOIN defaultdb.public.seed_vec@[0] AS tab_142430 ON (tab_142429._float8) = (tab_142430._float8),
			defaultdb.public.seed_vec@[0] AS tab_142431
		LIMIT
			26:::INT8;
```
<details><summary>Reproduce</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/roachtest)



See: [CI job to stress roachtests](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestStress)

<p>For the CI stress job, click the ellipsis (...) next to the Run button and fill in:
* Changes / Build branch: master
* Parameters / `env.TESTS`: `^sqlsmith/setup=seed-vec/setting=vec$`
* Parameters / `env.COUNT`: &lt;number of runs&gt;
</p>
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*sqlsmith/setup=seed-vec/setting=vec.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.sqlsmith/setup=seed-vec/setting=vec [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3281225&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3281225&tab=artifacts#/sqlsmith/setup=seed-vec/setting=vec) on master @ [847514dab6354d4cc4ccf7b2857487b32119fb37](https://github.com/cockroachdb/cockroach/commits/847514dab6354d4cc4ccf7b2857487b32119fb37):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/sqlsmith/setup=seed-vec/setting=vec/run_1
	sqlsmith.go:223,sqlsmith.go:252,test_runner.go:777: error: pq: inbox communication error: rpc error: code = Canceled desc = context canceled
		stmt:
		SELECT
			tab_74392.tableoid AS col_272259, tab_74391._int8 AS col_272260
		FROM
			defaultdb.public.seed_vec@[0] AS tab_74391,
			defaultdb.public.seed_vec@[0] AS tab_74392
			JOIN defaultdb.public.seed_vec@seed_vec__int8__float8__date_idx AS tab_74393 ON
					(tab_74392._int8) = (tab_74393._int2) AND (tab_74392._int2) = (tab_74393._int4),
			defaultdb.public.seed_vec@seed_vec__int8__float8__date_idx AS tab_74394
		WHERE
			tab_74394._bool
		ORDER BY
			tab_74393.tableoid ASC
		LIMIT
			19:::INT8;
```
<details><summary>Reproduce</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/roachtest)



See: [CI job to stress roachtests](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestStress)

<p>For the CI stress job, click the ellipsis (...) next to the Run button and fill in:
* Changes / Build branch: master
* Parameters / `env.TESTS`: `^sqlsmith/setup=seed-vec/setting=vec$`
* Parameters / `env.COUNT`: &lt;number of runs&gt;
</p>
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*sqlsmith/setup=seed-vec/setting=vec.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

roachtest.sqlsmith/setup=seed-vec/setting=vec [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3305136&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3305136&tab=artifacts#/sqlsmith/setup=seed-vec/setting=vec) on master @ [04a41e7915f4a89dcc1d0dbd92466c6adf79ec9f](https://github.com/cockroachdb/cockroach/commits/04a41e7915f4a89dcc1d0dbd92466c6adf79ec9f):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/sqlsmith/setup=seed-vec/setting=vec/run_1
	sqlsmith.go:223,sqlsmith.go:252,test_runner.go:777: error: pq: inbox communication error: rpc error: code = Canceled desc = context canceled
		stmt:
		SELECT
			tab_29557._timestamp AS col_107131,
			tab_29554._decimal AS col_107132,
			tab_29556._float8 AS col_107133,
			tab_29553._bool AS col_107134,
			tab_29556._float8 AS col_107135,
			tab_29557._bytes AS col_107136,
			tab_29557.tableoid AS col_107137,
			tab_29556._float8 AS col_107138,
			tab_29557.crdb_internal_mvcc_timestamp AS col_107139,
			tab_29553.tableoid AS col_107140,
			tab_29554._int8 AS col_107141,
			tab_29554.tableoid AS col_107142
		FROM
			defaultdb.public.seed_vec@seed_vec__int8__float8__date_idx AS tab_29553,
			defaultdb.public.seed_vec@[0] AS tab_29554,
			defaultdb.public.seed_vec AS tab_29555
			JOIN defaultdb.public.seed_vec AS tab_29556 ON
					(tab_29555._int8) = (tab_29556._int8)
					AND (tab_29555._date) = (tab_29556._date)
					AND (tab_29555._float8) = (tab_29556._float8),
			defaultdb.public.seed_vec@[0] AS tab_29557
		ORDER BY
			tab_29557._bool ASC, tab_29554._bytes DESC, tab_29553._bool
		LIMIT
			30:::INT8;
```
<details><summary>Reproduce</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/roachtest)



See: [CI job to stress roachtests](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestStress)

<p>For the CI stress job, click the ellipsis (...) next to the Run button and fill in:
* Changes / Build branch: master
* Parameters / `env.TESTS`: `^sqlsmith/setup=seed-vec/setting=vec$`
* Parameters / `env.COUNT`: &lt;number of runs&gt;
</p>
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*sqlsmith/setup=seed-vec/setting=vec.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/cmd/roachtest/tests/sqlsmith.go
+++ b/pkg/cmd/roachtest/tests/sqlsmith.go
@@ -12,6 +12,7 @@ package tests
 
 import (
 	"context"
+	gosql "database/sql"
 	"fmt"
 	"math/rand"
 	"os"
@@ -28,6 +29,7 @@ import (
 )
 
 func registerSQLSmith(r registry.Registry) {
+	const numNodes = 4
 	setups := map[string]sqlsmith.Setup{
 		"empty":       sqlsmith.Setups["empty"],
 		"seed":        sqlsmith.Setups["seed"],
@@ -102,7 +104,11 @@ func registerSQLSmith(r registry.Registry) {
 		setup := setupFunc(rng)
 		setting := settingFunc(rng)
 
-		conn := c.Conn(ctx, 1)
+		allConns := make([]*gosql.DB, 0, numNodes)
+		for node := 1; node <= numNodes; node++ {
+			allConns = append(allConns, c.Conn(ctx, node))
+		}
+		conn := allConns[0]
 		t.Status("executing setup")
 		t.L().Printf("setup:\n%s", setup)
 		if _, err := conn.Exec(setup); err != nil {
@@ -132,17 +138,11 @@ func registerSQLSmith(r registry.Registry) {
 		// other setup queries have already completed, including the smither
 		// instantiation (otherwise, the setup might fail because of the
 		// injected panics).
-		if rng.Float64() < 0.5 {
-			// TODO(yuzefovich): at the moment we're only injecting panics with
-			// 50% probability in order to test the hypothesis that this panic
-			// injection is the root cause of the inbox communication errors we
-			// have been seeing sporadically.
-			injectPanicsStmt := "SET testing_vectorize_inject_panics=true;"
-			if _, err := conn.Exec(injectPanicsStmt); err != nil {
-				t.Fatal(err)
-			}
-			logStmt(injectPanicsStmt)
+		injectPanicsStmt := "SET testing_vectorize_inject_panics=true;"
+		if _, err := conn.Exec(injectPanicsStmt); err != nil {
+			t.Fatal(err)
 		}
+		logStmt(injectPanicsStmt)
 
 		t.Status("smithing")
 		until := time.After(t.Spec().(*registry.TestSpec).Timeout / 2)
@@ -222,11 +222,6 @@ func registerSQLSmith(r registry.Registry) {
 						logStmt(stmt)
 						t.Fatalf("error: %s\nstmt:\n%s;", err, stmt)
 					}
-				} else if strings.Contains(es, "communication error") {
-					// A communication error can be because
-					// a non-gateway node has crashed.
-					logStmt(stmt)
-					t.Fatalf("error: %s\nstmt:\n%s;", err, stmt)
 				} else if strings.Contains(es, "Empty statement returned by generate") ||
 					stmt == "" {
 					// Either were unable to generate a statement or
@@ -237,12 +232,15 @@ func registerSQLSmith(r registry.Registry) {
 				// frequently (due to sqlsmith not crafting
 				// executable queries 100% of the time) and are
 				// never interesting.
+				// TODO(yuzefovich): reevaluate this assumption.
 			}
 
-			// Ping the gateway to make sure it didn't crash.
-			if err := conn.PingContext(ctx); err != nil {
-				logStmt(stmt)
-				t.Fatalf("ping: %v\nprevious sql:\n%s;", err, stmt)
+			// Ping all nodes to make sure they didn't crash.
+			for idx, c := range allConns {
+				if err := c.PingContext(ctx); err != nil {
+					logStmt(stmt)
+					t.Fatalf("ping node %d: %v\nprevious sql:\n%s;", idx+1, err, stmt)
+				}
 			}
 		}
 	}
@@ -252,7 +250,7 @@ func registerSQLSmith(r registry.Registry) {
 			Name: fmt.Sprintf("sqlsmith/setup=%s/setting=%s", setup, setting),
 			// NB: sqlsmith failures should never block a release.
 			Owner:   registry.OwnerSQLQueries,
-			Cluster: r.MakeClusterSpec(4),
+			Cluster: r.MakeClusterSpec(numNodes),
 			Timeout: time.Minute * 20,
 			Run: func(ctx context.Context, t test.Test, c cluster.Cluster) {
 				runSQLSmith(ctx, t, c, setup, setting)

--- a/pkg/sql/colflow/colrpc/inbox.go
+++ b/pkg/sql/colflow/colrpc/inbox.go
@@ -323,7 +323,7 @@ func (i *Inbox) Next() coldata.Batch {
 			}
 			// Note that here err can be stream's context cancellation.
 			// Regardless of the cause we want to propagate such an error as
-			// expected on in all cases so that the caller could decide on how
+			// expected one in all cases so that the caller could decide on how
 			// to handle it.
 			err = pgerror.Newf(pgcode.InternalConnectionFailure, "inbox communication error: %s", err)
 			i.errCh <- err

